streams/test-utils/src/main/java/org/apache/kafka/streams/processor/internals/TestDriverProducer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.utils.LogContext;
+
+import java.util.Map;
+
+public class TestDriverProducer extends StreamsProducer {
+
+    public TestDriverProducer(final Producer<byte[], byte[]> producer,
+                              final String applicationId,
+                              final LogContext logContext) {
+        super(producer, applicationId, logContext);
+    }
+
+    public void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets) throws ProducerFencedException {
```

Comment:
Just added to give public access to `commitTransaction()` to TTD (it's more elegant than making `StreamsProducer#commitTransaction` public IMHO)

Reply:
Why making `commitTransaction` is less elegant? I thought that was fine since `StreamsProducer` is inside the internals package anyways? In fact, in TTD we have access to InternalTopologyBuilder accessing it functions (we used to also have a wrapper of InternalTopologyBuilder which we removed later) so I thought that was the agreed pattern.

Reply:
It's obviously subjective -- personally, even if something is internal, we should not just declare stuff as `public` but try to keep it to a minimum to follow the idea of encapsulation (not always possible). If you want me to remove this class and make the method `public` I can do it in a follow up PR. Not sure if we have an agreed pattern, though.

Reply:
Cool, in that sense let's just keep it then -- do not add it in one PR and remove it immediately in the next.